### PR TITLE
Mark attendance command v1.3

### PR DESCRIPTION
**Pull request by joosxi**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX_NUMBERS;
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]:
```suggestion
import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX_NUMBER;
```

Comment (e1010101) [reply src/main/java/seedu/address/logic/commands/AddCommand.java]: + for the rest as well

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Marks the attendance of persons in address book with studentClass and indexNumber corresponding to the
+ * user's arguments.
+ */
+public class MarkAttendanceCommand extends Command {
+
+    public static final String COMMAND_WORD = "markAtt";
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: might as well use markAttendance here?

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Marks attendance of all students in the "
+                                               + "specified class with index numbers matching the user input\n"
+                                               + "Parameters: d/DATE (DD/MM/YYYY) "
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: Use we also support other date formats! See the DateUtil class!

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                                               + "c/CLASS (only one class should be entered)"
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: + class should not be blank

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                                               + "ind/INDEX_NUMBERS (index numbers of students, separated by spaces)\n"
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: "...of students that are ABSENT..."

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                                               + "Example: " + COMMAND_WORD + " d/26/10/2022 c/1A ind/1 5 8 10\n"
+                                               + "Example: " + COMMAND_WORD + " d/05/08/2022 c/1B ind/4 7 9 10";
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: One example is enough!

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java (hunk 5 above)
Comment (e1010101) [reply src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: + maybe use "index" for the prefix! Not much time saved by shortening by 2 letters

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+    private static final String MESSAGE_MARK_ATTENDANCE_SUCCESS = "Attendance for students have been marked.";
+    private static final String MESSAGE_MARK_ATTENDANCE_NONE = "No student's attendance has been marked.";
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: Use case for this? If no students are input (all students are PRESENT!) then attendance is still marked (as all "1")

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    private String date;
+    private String studentClass;
+    private List<String> indexNumbers;
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: Why not List<IndexNumber>?

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+     */
+    public CommandResult execute(Model model) {
+        boolean anyStudentsAttendanceMarked = false;
+        String markedAttendance = "";
+        requireNonNull(model);
+        // Now we're iterating through the whole list of saved people in Watsons and checking against their
+        // indexNumber and studentClass to check if they should be marked attendance. Not the most efficient.
+
+        //Predicate<Person> personIsInClassPredicate = model.createPersonIsInClassPredicate(studentClass);
+        //model.updateFilteredPersonList(personIsInClassPredicate);
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: Good comment!

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
+
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java]: Possible to optimize the algorithm by using a better predicate here, KIV for v1.4!

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 11]
```diff
@@ -19,5 +19,9 @@ public class CliSyntax {
     public static final Prefix PREFIX_SUBJECT = new Prefix("s/");
     public static final Prefix PREFIX_GRADE = new Prefix("g/");
 
+    // Fields for marking attendance
+    public static final Prefix PREFIX_DATE = new Prefix("d/");
+    public static final Prefix PREFIX_INDEX_NUMBERS = new Prefix("ind/");
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java]: Consider using index/ for the prefix for index numbers, see earlier comments!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 12]
```diff
@@ -137,6 +137,11 @@ public void sortListByGrade(boolean isInAscending) {
         this.addressBook.sortByGrade(isInAscending);
     }
 
+    @Override
+    public Predicate<Person> createPersonIsInClassPredicate(String studentClass) {
+        return person -> person.getStudentClass().isSameClass(studentClass);
+    }
+
```
Comment (e1010101) [inline src/main/java/seedu/address/model/ModelManager.java]: Can use this in the MarkAttendanceCommand right?

File: src/main/java/seedu/address/logic/parser/MarkAttendanceCommandParser.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+            throw new ParseException(
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkAttendanceCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args, PREFIX_DATE, PREFIX_STUDENTCLASS, PREFIX_INDEX_NUMBERS);
+
+        String date = argMultimap.getValue(PREFIX_DATE).get();
+        String studentClass = argMultimap.getValue(PREFIX_STUDENTCLASS).get();
+        List<String> indexNumbers = argMultimap.getAllValues(PREFIX_INDEX_NUMBERS);
```
Comment (e1010101) [inline src/main/java/seedu/address/logic/parser/MarkAttendanceCommandParser.java]: Why not List<IndexNumber>?

**Review by e1010101 — CHANGES_REQUESTED** [review]
Cool!

File: src/main/java/seedu/address/model/ModelManager.java (hunk 12 above)
Comment (joosxi) [reply src/main/java/seedu/address/model/ModelManager.java]: Ya I tried but then it didn't work leh that's why I commented out some code using this in the MarkAttendanceCommand class

**Review by jgyj123 — APPROVED** [review]
LGTM

**Pull request merged**